ios/browser/api/sync/brave_sync_api.h

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_SYNC_BRAVE_SYNC_API_H_
+#define BRAVE_IOS_BROWSER_API_SYNC_BRAVE_SYNC_API_H_
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+OBJC_EXPORT
+@interface BraveSyncDeviceObserver: NSObject
+- (instancetype)initWithCallback:(void(^)())onDeviceInfoChanged;
+@end
+
+OBJC_EXPORT
+@interface BraveSyncServiceObserver: NSObject
+- (instancetype)initWithCallback:(void(^)())onSyncServiceStateChanged;
+@end
+
+OBJC_EXPORT
+@interface BraveSyncAPI : NSObject
+
+@property(class, readonly, strong) BraveSyncAPI *sharedSyncAPI NS_SWIFT_NAME(shared);
+
+- (bool)setSyncEnabled:(bool)enabled;
+
+- (bool)resetSync;
+
+- (NSString *)getSyncCode;
+
+// returns false is sync is already configured or if the sync code is invalid
+- (bool)setSyncCode:(NSString *)syncCode;
+
+- (NSString *)syncCodeFromHexSeed:(NSString *)hexSeed;
+
+- (nullable UIImage *)getQRCodeImage:(CGSize)size;
+
+- (nullable NSString *)getDeviceListJSON;
```

Comment:
Naming nit (in general of this file): Obj-C usually doesn't have `get` prefixed in the name